source/extensions/common/aws/signer_base_impl.h

```diff
@@ -24,6 +24,23 @@ class SignatureHeaderValues {
 
 using SignatureHeaders = ConstSingleton<SignatureHeaderValues>;
 
+class SignatureQueryParameterValues {
+public:
+  // Query string parameters require camel case
+  static constexpr char AmzAlgorithm[] = "X-Amz-Algorithm";
+  static constexpr char AmzCredential[] = "X-Amz-Credential";
+  static constexpr char AmzDate[] = "X-Amz-Date";
+  static constexpr char AmzRegionSet[] = "X-Amz-Region-Set";
+  static constexpr char AmzSecurityToken[] = "X-Amz-Security-Token";
+  static constexpr char AmzSignature[] = "X-Amz-Signature";
+  static constexpr char AmzSignedHeaders[] = "X-Amz-SignedHeaders";
+  static constexpr char AmzExpires[] = "X-Amz-Expires";
+  // Expiration time of query parameter request, in seconds
+  const uint16_t DefaultExpiration = 5;
```

Comment:
`constexpr abs::string_view` is prefered.

` static constexpr uint16_t DefaultExpiration = 5;`